Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -1503,6 +1561,60 @@ end Point
 
 end Affine
 
+section Map
+
+lemma comp_fin3 {S} (f : R → S) (X Y Z : R) : f ∘ ![X, Y, Z] = ![f X, f Y, f Z] := by
+  ext i; fin_cases i <;> rfl
+
+variable {S : Type*} [CommRing S] (f : R →+* S) (P Q : Fin 3 → R)
+
+protected lemma map_smul (u : R) : f ∘ (u • P) = f u • (f ∘ P) := by
+  ext i; fin_cases i <;> simp [smul_fin3]
+
+@[simp] lemma map_addZ : addZ (f ∘ P) (f ∘ Q) = f (addZ P Q) := by simp [addZ]
```

Comment:
I think the direction should be `f (addZ P Q) = addZ (f ∘ P) (f ∘ Q)` based on `map` lemmas throughout mathlib, and I don't think you should make this `simp` in that case. Similarly for the other lemmas below this.

Reply:
This is a good point; it seems I'm only using `map_addXYZ` and `map_dblXYZ` in the reverse direction, so it's probably good to reverse the direction and keep the `@[simp]`.

Reply:
In https://github.com/leanprover-community/mathlib4/pull/13846/commits/1a5e80ac9c8729d761e69b44ab137535bb622b20 I switched the direction of the `map` lemmas except for those about `polynomial(X,Y,Z)`. Now `map_polynomial(X,Y,Z)` has `WeierstrassCurve.map` on LHS, in contrast to the other `map` lemmas, but they agree with the affine lemmas, and since they are not marked `@[simp]` now I think it's okay. I think their names are acceptable, since their are plenty of lemmas following this naming convention like [Prod.map_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Prod/Basic.html#Prod.map_injective).

Reply:
Hmm should I really have switched the directions? Now these `simp` lemmas are contrary to the `simp` direction of [WeierstrassCurve.map_a₁](https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.html#WeierstrassCurve.map_a%E2%82%81) etc.

Reply:
This is why I'm hesitant on which direction simp should be. I don't know if the simp direction of `map_a*` is correct.

Reply:
Well, those map_a* are generated by @[simps] so it would be hard to argue the direction isn't correct. I think I'll revert the commit ...